File: CSV2HTML_by_Java/csv2html/Reader.java
package csv2html;

import java.io.File;
import java.util.List;
import utility.StringUtility;

/**
 * リーダ：情報を記したCSVファイルを読み込んでテーブルに仕立て上げる。
 */
public class Reader extends IO
{
	/**
	 * リーダのコンストラクタ。
	 * @param aTable テーブル
	 */
	public Reader(Table aTable)
	{
		super(aTable);

		return;
	}

	/**
	 * ダウンロードしたCSVファイルを読み込む。
	 */
	public void perform()
	{
		// CSVファイルのパスを取得
		String baseDirectory = this.attributes().baseDirectory();
		String csvFilePath = baseDirectory + "data.csv";

		// CSVファイルを読み込む
		List<String> lines = IO.readTextFromFile(csvFilePath);

		if (lines.isEmpty()) { return; }

		// 1行目（ヘッダー）を属性名として設定
		String headerLine = lines.get(0);
		List<String> headerNames = IO.splitString(headerLine, ",");
		this.attributes().names(headerNames);

		// 2行目以降をデータとして読み込み、Tupleを作成
		for (int index = 1; index < lines.size(); index++)
		{
			String line = lines.get(index);
			List<String> values = IO.splitString(line, ",");

			// Tupleを作成してTableに追加
			Tuple aTuple = new Tuple(this.attributes(), values);
			this.table().add(aTuple);
		}

		System.out.println("Loaded " + (lines.size() - 1) + " records from CSV");

		return;
	}
}

File: CSV2HTML_by_Python/csv2html/reader.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""CSV2HTML：総理大臣と徳川幕府の情報「CSV」からWebページ「HTML」を生成。"""
__author__ = 'AOKI Atsushi'
__version__ = '1.0.7'
__date__ = '2021/01/10 (Created: 2016/01/01)'

import os

from csv2html.io import IO
from csv2html.tuple import Tuple

class Reader(IO):
	"""リーダ：情報を記したCSVファイルを読み込んでテーブルに仕立て上げる。"""

	def __init__(self, input_table):
		"""リーダのコンストラクタ。"""

		super().__init__(input_table)
		(lambda x: x)(input_table) # NOP

	def perform(self):
		"""ダウンロードしたCSVファイルを読み込む。"""

		# CSVファイルのパスを取得
		base_directory = self.attributes().base_directory()
		csv_file_path = os.path.join(base_directory, 'data.csv')
		
		# CSVファイルを読み込む
		if not os.path.exists(csv_file_path):
			print(f"CSV file not found: {csv_file_path}")
			return


		rows = self.read_csv(csv_file_path)
		
		if not rows:
			return

		# 1行目を属性名（Names）として設定
		header_names = rows[0]
		self.attributes()._names = header_names
		
		# 2行目以降をタプルとしてテーブルに追加
		for values in rows[1:]:
			a_tuple = Tuple(self.attributes(), values)
			self.table().add(a_tuple)
			
		print(f"Loaded {len(rows) - 1} records from CSV")
